src/connectStoreController.ts

```diff
@@ -62,18 +62,22 @@ export default function (
 
 function fetch(limit: (fn: () => PromiseLike<object>) => Promise<object>, url: string, body: object): Promise<object | undefined> { // tslint:disable-line
   return limit(async () => {
-    const response = await got(url, {
-      body: JSON.stringify(body),
-      headers: {'Content-Type': 'application/json'},
-      method: 'POST',
-      retries: () => {
-        return 100
-      },
-    })
-    if (!response.body) {
-      return undefined
+    try {
+      const response = await got(url, {
+        body: JSON.stringify(body),
+        headers: {'Content-Type': 'application/json'},
+        method: 'POST',
+        retries: () => {
+          return 100
+        },
+      })
+      if (!response.body) {
+        return undefined
+      }
+      return JSON.parse(response.body)
+    } catch (e) {
+      throw new Error(JSON.parse(e.response.body))
```

Comment:
the body contains only the message? I know that specific error contains more details. see https://github.com/pnpm/npm-resolver/blob/cf323ffb29ad898ca002d089afbd35e855c3c6c5/src/pickPackage.ts#L146-L153